app/scripts/modules/core/pipeline/config/validation/requiredField.validator.ts

```diff
@@ -0,0 +1,53 @@
+import {module} from 'angular';
+
+import {
+  PIPELINE_CONFIG_VALIDATOR, IStageOrTriggerTypeConfig, IStageOrTriggerValidator, IValidatorConfig,
+  PipelineConfigValidator
+} from './pipelineConfig.validator';
+import {IStage} from 'core/domain/IStage';
+import {ITrigger} from 'core/domain/ITrigger';
+import {IPipeline} from 'core/domain/IPipeline';
+
+export interface IRequiredFieldValidationConfig extends IValidatorConfig {
+  fieldName: string;
+  fieldLabel?: string;
+  message?: string;
+}
+
+export class RequiredFieldValidator implements IStageOrTriggerValidator {
+
+  public validate(pipeline: IPipeline, stage: IStage | ITrigger, validationConfig: IRequiredFieldValidationConfig, config: IStageOrTriggerTypeConfig): string {
+    if (pipeline.strategy === true && ['cluster', 'regions', 'zones', 'credentials'].includes(validationConfig.fieldName)) {
+      return null;
+    }
+
+    let fieldLabel: string = validationConfig.fieldLabel || validationConfig.fieldName;
+    fieldLabel = fieldLabel.charAt(0).toUpperCase() + fieldLabel.substr(1);
+    const validationMessage = validationConfig.message || `<strong>${fieldLabel}</strong> is a required field for ${config.label} stages.`;
+    let field: any = stage,
+      parts = validationConfig.fieldName.split('.'),
+      fieldNotFound = false;
+
+    parts.forEach((part) => {
```

Comment:
Could this be simplified as `_.get(field, validationConfig.fieldName)` and a check if the result is undefined?